packages/devtools_app/lib/src/extensions/extension_service.dart

```diff
@@ -14,17 +15,36 @@ class ExtensionService extends DisposableController
       _availableExtensions;
   final _availableExtensions = ValueNotifier<List<DevToolsExtensionConfig>>([]);
 
-  void initialize() {
-    addAutoDisposeListener(serviceManager.connectedState, () {
-      _refreshAvailableExtensions();
+  Future<void> initialize() async {
+    await maybeRefreshExtensions();
+    addAutoDisposeListener(serviceManager.connectedState, () async {
```

Comment:
Can this just be `autoDisposeListener(serviceManager.connectedState, maybeRefreshExtensions)`?